Qt5 + cmake doesn't seem to work at all

**TL;DR: MXE+Qt5+cmake doesn't seem to work.**

Hi, I've been trying for quite a while to make sigrok (www.sigrok.org) Windows packages using Qt5 + MXE, without any success unfortunately (our current packages use Qt4, which has also been a challenge to implement as well, but that's somewhat working at least).

For our purposes we're depending on a number of specific things that need to work together: 

- MXE target "i686-w64-mingw32.static.posix" (we need C++11 + threads etc.)
  (and static builds of everything else in MXE and outside MXE, of course)
- cmake (used by our PulseView frontend)
- Qt5 (we cannot and don't want to remain stuck with Qt4 forever, for many different reasons, including features from Qt5 we want, and bugfixes from Qt5 we need)

Within Qt5 we need the following: Qt5Core, Qt5Gui, Qt5Svg and various static plugins (directly or indirectly) such as qsvg, QWindowsIntegrationPlugin and such.

I've spent countless days trying to get a successful build, but I didn't get anywhere near a successful build that will also work at runtime.

So my question is whether I'm doing something wrong or whether there are bugs in MXE e.g. regarding i686-w64-mingw32.static.posix+Qt5+cmake that would prevent this from working?

In order to get something reproducible that demonstrates the problem without too many dependencies on all kinds of other stuff we do in sigrok, I've created this demo repo here:

https://github.com/uwehermann/qt5helloworld

It contains a simple project that I generated in qtcreator on Linux which basically just shows an empty window. I added the simplest-possible CMakeLists.txt file for buiding it, and two shell scripts to demonstrate exactly how I'm building.

- The Linux build script works fine, I get an executable and I can run the executable successfully.
- The MXE script doesn't yield a successful build, there are many linker errors (this is with MXE git as of today, only "make qt5", no Qt4 in the MXE paths anywhere).

I've tried all kinds of things in CMakeLists.txt, the source code itself, random MXE patching, nothing has worked sufficiently well so far.

Can someone please verify that the repo I provided does indeed build fine on Linux and doesn't build with MXE?

If yes, any patches against MXE and/or that repo to make this simple project build and run successfully would be really appreciated. It would at least be a basis for further work (I'm pretty sure the static Qt5 plugins will be the next major issue causing headaches after that).

Please let me know if you need any more info. Thanks!

After going through the discussion here I'm now linking my app like this:

`target_link_libraries(${BINARY_NAME} Qt5::Widgets Qt5::Xml Qt5::QWindowsIntegrationPlugin)`

But I get a lot of stuff like this:

```

/opt/mxe/usr/i686-w64-mingw32.static/qt5/plugins/platforms/libqwindows.a(qwindowsintegration.o):qwindowsintegration.cpp:(.text+0xcd): undefined reference to `QWindowsFontDatabase::QWindowsFontDatabase()'
/opt/mxe/usr/i686-w64-mingw32.static/qt5/plugins/platforms/libqwindows.a(qwindowsintegration.o):qwindowsintegration.cpp:(.text+0xf1): undefined reference to `vtable for QWindowsFontDatabaseFT'
/opt/mxe/usr/i686-w64-mingw32.static/qt5/plugins/platforms/libqwindows.a(qwindowsintegration.o):qwindowsintegration.cpp:(.text+0x151): undefined reference to `QWindowsFontDatabase::fontSmoothingGamma()'
/opt/mxe/usr/i686-w64-mingw32.static/qt5/plugins/platforms/libqwindows.a(qwindowsintegration.o):qwindowsintegration.cpp:(.text+0x2fc): undefined reference to `QWindowsGuiEventDispatcher::QWindowsGuiEventDispatcher(QObject*)'
/opt/mxe/usr/i686-w64-mingw32.static/qt5/plugins/platforms/libqwindows.a(qwindowsintegration.o):qwindowsintegration.cpp:(.text+0x1ce6): undefined reference to `QWindowsFontDatabase::setFontOptions(unsigned int)'
/opt/mxe/usr/i686-w64-mingw32.static/qt5/plugins/platforms

```

I also have:

```
#include <QtPlugin>
Q_IMPORT_PLUGIN(QWindowsIntegrationPlugin)
```

And

`add_definitions(-DQT_STATICPLUGIN)`

There are definitely some progress made. Trying to build Qt5 application generated in Qt Creator using template "Qt Widgets Application" with current mxe master branch (CMake 3.15.4, Qt 5.14.0) results in just few linking errors:

```
# x86_64-w64-mingw32.static-cmake /host/home/artem/development/test/mxe_qt_static_app
== Using MXE wrapper: /mxe/usr/bin/x86_64-w64-mingw32.static-cmake
     - cmake version 3.15.4
     - warnings for unused CMAKE_POLICY_DEFAULT variables can be ignored
== Using MXE toolchain: /mxe/usr/x86_64-w64-mingw32.static/share/cmake/mxe-conf.cmake
== Using MXE runresult: /mxe/usr/share/cmake/modules/TryRunResults.cmake
== Adding "-DCMAKE_BUILD_TYPE=Release"
loading initial cache file /mxe/usr/share/cmake/modules/TryRunResults.cmake
-- The CXX compiler identification is GNU 5.5.0
-- Check for working CXX compiler: /mxe/usr/x86_64-pc-linux-gnu/bin/x86_64-w64-mingw32.static-g++
-- Check for working CXX compiler: /mxe/usr/x86_64-pc-linux-gnu/bin/x86_64-w64-mingw32.static-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0017
    CMAKE_POLICY_DEFAULT_CMP0020


-- Build files have been written to: /work/build-HardwareTestApp-mxe-x86_64-w64-mingw32
# make
Scanning dependencies of target mxe_qt_static_app_autogen
[ 10%] Automatic MOC and UIC for target mxe_qt_static_app
[ 10%] Built target mxe_qt_static_app_autogen
Scanning dependencies of target mxe_qt_static_app
[ 20%] Building CXX object CMakeFiles/mxe_qt_static_app.dir/mxe_qt_static_app_autogen/mocs_compilation.cpp.obj
[ 30%] Building CXX object CMakeFiles/mxe_qt_static_app.dir/main.cpp.obj
[ 40%] Building CXX object CMakeFiles/mxe_qt_static_app.dir/widget.cpp.obj
[ 50%] Building CXX object CMakeFiles/mxe_qt_static_app.dir/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/cmake/Qt5Gui/Qt5Gui_QGifPlugin_Import.cpp.obj
[ 60%] Building CXX object CMakeFiles/mxe_qt_static_app.dir/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/cmake/Qt5Gui/Qt5Gui_QICOPlugin_Import.cpp.obj
[ 70%] Building CXX object CMakeFiles/mxe_qt_static_app.dir/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/cmake/Qt5Gui/Qt5Gui_QJpegPlugin_Import.cpp.obj
[ 80%] Building CXX object CMakeFiles/mxe_qt_static_app.dir/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/cmake/Qt5Gui/Qt5Gui_QWindowsIntegrationPlugin_Import.cpp.obj
[ 90%] Building CXX object CMakeFiles/mxe_qt_static_app.dir/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/cmake/Qt5Widgets/Qt5Widgets_QWindowsVistaStylePlugin_Import.cpp.obj
[100%] Linking CXX executable mxe_qt_static_app.exe
/mxe/usr/bin/x86_64-w64-mingw32.static-ld: cannot find -ljasper
/mxe/usr/bin/x86_64-w64-mingw32.static-ld: cannot find -lmng
/mxe/usr/bin/x86_64-w64-mingw32.static-ld: cannot find -ltiff
/mxe/usr/bin/x86_64-w64-mingw32.static-ld: cannot find -lwebpdemux
/mxe/usr/bin/x86_64-w64-mingw32.static-ld: cannot find -lwebpmux
/mxe/usr/bin/x86_64-w64-mingw32.static-ld: cannot find -llcms2
/mxe/usr/bin/x86_64-w64-mingw32.static-ld: cannot find -lwebp
/mxe/usr/bin/x86_64-w64-mingw32.static-ld: cannot find -llzma
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/mxe_qt_static_app.dir/build.make:224: mxe_qt_static_app.exe] Error 1
make[1]: *** [CMakeFiles/Makefile2:77: CMakeFiles/mxe_qt_static_app.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```
My words are such optimistic because trying to build a little more complex app (i.e. with more dependencies) gives same result.
Not sure whether should I try apply #1645 because it seems to be out of date. Why I'm tried to make things using latest mxe version is just because I was driven by hope that closed [longstanding qt bug](https://bugreports.qt.io/browse/QTBUG-38913) will finally resolve issue. Well, it did progress (note cmake output above where it automates integrating static plugins). Maybe it's just not Qt fail, but rather MXE should be updated accordingly...

... but application using QtSerialPort fails to link:
```
/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/libQt5SerialPort.a(qserialportinfo_win.o):qserialportinfo_win.cpp:(.text$_ZL24deviceInstanceIdentifierm+0x3f): undefined reference to `__imp_CM_Get_Device_IDW'
/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/libQt5SerialPort.a(qserialportinfo_win.o):qserialportinfo_win.cpp:(.text$_ZL22deviceRegistryPropertyPvP16_SP_DEVINFO_DATAm+0x79): undefined reference to `__imp_SetupDiGetDeviceRegistryPropertyW'
/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/libQt5SerialPort.a(qserialportinfo_win.o):qserialportinfo_win.cpp:(.text$_ZN15QSerialPortInfo14availablePortsEv+0x7b): undefined reference to `__imp_SetupDiGetClassDevsW'
/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/libQt5SerialPort.a(qserialportinfo_win.o):qserialportinfo_win.cpp:(.text$_ZN15QSerialPortInfo14availablePortsEv+0x108): undefined reference to `__imp_SetupDiEnumDeviceInfo'
/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/libQt5SerialPort.a(qserialportinfo_win.o):qserialportinfo_win.cpp:(.text$_ZN15QSerialPortInfo14availablePortsEv+0x135): undefined reference to `__imp_SetupDiOpenDevRegKey'
/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/libQt5SerialPort.a(qserialportinfo_win.o):qserialportinfo_win.cpp:(.text$_ZN15QSerialPortInfo14availablePortsEv+0x4d3): undefined reference to `__imp_CM_Get_DevNode_Status'
/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/libQt5SerialPort.a(qserialportinfo_win.o):qserialportinfo_win.cpp:(.text$_ZN15QSerialPortInfo14availablePortsEv+0x865): undefined reference to `__imp_CM_Get_Parent'
/mxe/usr/x86_64-w64-mingw32.static/qt5/lib/libQt5SerialPort.a(qserialportinfo_win.o):qserialportinfo_win.cpp:(.text$_ZN15QSerialPortInfo14availablePortsEv+0x941): undefined reference to `__imp_SetupDiDestroyDeviceInfoList'
```

--- a/src/qtbase-1-fixes.patch
+++ b/src/qtbase-1-fixes.patch
@@ -63,3 +63,102 @@ index 19312d245d..0164f9b49b 100644
 -- 
 2.20.1
 
+
+From b8fd37d8cd7642599b3b022e4d8fa1aff8f68a1e Mon Sep 17 00:00:00 2001
+From: Artem Pisarenko <artem.k.pisarenko@gmail.com>
+Date: Thu, 16 Jan 2020 14:47:00 +0600
+Subject: [PATCH] Fix win32 support for linking against static builds in CMake
+
+CMake mkspec feature sets CMAKE_PRL_FILE_LOCATION_* values inconsistent
+with actual .prl file names being generated/installed, causing
+dependency extraction mechanism in Qt5BasicConfig.cmake.in to not work
+(function _qt5_$${CMAKE_MODULE_NAME}_process_prl_file silently fails
+to find file at given location and skips filling libs deps).
+
+[ChangeLog][CMake][Windows] Fixed linking with Qt static build
+
+Fixes: QTBUG-81401
+Change-Id: I5861cc0c42163d898ba55ad83cbad1994dcb1db2
+---
+ mkspecs/features/create_cmake.prf | 41 +++++++++++++--------------------------
+ 1 file changed, 13 insertions(+), 28 deletions(-)
+
+diff --git a/mkspecs/features/create_cmake.prf b/mkspecs/features/create_cmake.prf
+index 346fbf2..0d93597 100644
+--- a/mkspecs/features/create_cmake.prf
++++ b/mkspecs/features/create_cmake.prf
+@@ -212,21 +212,17 @@ contains(CONFIG, plugin) {
+     CMAKE_PLUGIN_TYPE_ESCAPED = $$replace(PLUGIN_TYPE, [-/], _)
+ 
+     win32 {
++        CMAKE_PRL_FILE_LOCATION_RELEASE = $$PLUGIN_TYPE/$${CMAKE_QT_STEM}.prl
++        CMAKE_PRL_FILE_LOCATION_DEBUG = $$PLUGIN_TYPE/$${CMAKE_QT_STEM}d.prl
+         isEmpty(CMAKE_STATIC_TYPE) {
+             CMAKE_PLUGIN_LOCATION_RELEASE = $$PLUGIN_TYPE/$${CMAKE_QT_STEM}.dll
+             CMAKE_PLUGIN_LOCATION_DEBUG = $$PLUGIN_TYPE/$${CMAKE_QT_STEM}d.dll
+-            CMAKE_PRL_FILE_LOCATION_RELEASE = $$PLUGIN_TYPE/$${CMAKE_QT_STEM}.prl
+-            CMAKE_PRL_FILE_LOCATION_DEBUG = $$PLUGIN_TYPE/$${CMAKE_QT_STEM}d.prl
+         } else:mingw {
+             CMAKE_PLUGIN_LOCATION_RELEASE = $$PLUGIN_TYPE/lib$${CMAKE_QT_STEM}.a
+             CMAKE_PLUGIN_LOCATION_DEBUG = $$PLUGIN_TYPE/lib$${CMAKE_QT_STEM}d.a
+-            CMAKE_PRL_FILE_LOCATION_RELEASE = $$PLUGIN_TYPE/lib$${CMAKE_QT_STEM}.prl
+-            CMAKE_PRL_FILE_LOCATION_DEBUG = $$PLUGIN_TYPE/lib$${CMAKE_QT_STEM}d.prl
+         } else {                         # MSVC static
+             CMAKE_PLUGIN_LOCATION_RELEASE = $$PLUGIN_TYPE/$${CMAKE_QT_STEM}.lib
+             CMAKE_PLUGIN_LOCATION_DEBUG = $$PLUGIN_TYPE/$${CMAKE_QT_STEM}d.lib
+-            CMAKE_PRL_FILE_LOCATION_RELEASE = $$PLUGIN_TYPE/$${CMAKE_QT_STEM}.prl
+-            CMAKE_PRL_FILE_LOCATION_DEBUG = $$PLUGIN_TYPE/$${CMAKE_QT_STEM}d.prl
+         }
+     } else {
+         mac {
+@@ -316,36 +312,25 @@ mac {
+     CMAKE_LIB_FILE_LOCATION_DEBUG = $${CMAKE_QT_STEM}d.dll
+     CMAKE_LIB_FILE_LOCATION_RELEASE = $${CMAKE_QT_STEM}.dll
+ 
++    !isEmpty(CMAKE_STATIC_TYPE) {
++        CMAKE_STATIC_WINDOWS_BUILD = "true"
++
++        CMAKE_PRL_FILE_LOCATION_DEBUG = $${CMAKE_QT_STEM}d.prl
++        CMAKE_PRL_FILE_LOCATION_RELEASE = $${CMAKE_QT_STEM}.prl
++    }
++
+     mingw {
+         CMAKE_WINMAIN_FILE_LOCATION_DEBUG = libqtmain$${QT_LIBINFIX}d.a
+         CMAKE_WINMAIN_FILE_LOCATION_RELEASE = libqtmain$${QT_LIBINFIX}.a
+ 
+-        !isEmpty(CMAKE_STATIC_TYPE) {
+-            CMAKE_STATIC_WINDOWS_BUILD = "true"
+-            CMAKE_IMPLIB_FILE_LOCATION_DEBUG = lib$${CMAKE_QT_STEM}d.a
+-            CMAKE_IMPLIB_FILE_LOCATION_RELEASE = lib$${CMAKE_QT_STEM}.a
+-
+-            CMAKE_PRL_FILE_LOCATION_DEBUG = lib$${CMAKE_QT_STEM}d.prl
+-            CMAKE_PRL_FILE_LOCATION_RELEASE = lib$${CMAKE_QT_STEM}.prl
+-        } else {
+-            CMAKE_IMPLIB_FILE_LOCATION_DEBUG = lib$${CMAKE_QT_STEM}d.a
+-            CMAKE_IMPLIB_FILE_LOCATION_RELEASE = lib$${CMAKE_QT_STEM}.a
+-        }
++        CMAKE_IMPLIB_FILE_LOCATION_DEBUG = lib$${CMAKE_QT_STEM}d.a
++        CMAKE_IMPLIB_FILE_LOCATION_RELEASE = lib$${CMAKE_QT_STEM}.a
+     } else {
+         CMAKE_WINMAIN_FILE_LOCATION_DEBUG = qtmain$${QT_LIBINFIX}d.lib
+         CMAKE_WINMAIN_FILE_LOCATION_RELEASE = qtmain$${QT_LIBINFIX}.lib
+ 
+-        !isEmpty(CMAKE_STATIC_TYPE) {
+-            CMAKE_STATIC_WINDOWS_BUILD = "true"
+-            CMAKE_IMPLIB_FILE_LOCATION_DEBUG = $${CMAKE_QT_STEM}d.lib
+-            CMAKE_IMPLIB_FILE_LOCATION_RELEASE = $${CMAKE_QT_STEM}.lib
+-
+-            CMAKE_PRL_FILE_LOCATION_DEBUG = $${CMAKE_QT_STEM}d.prl
+-            CMAKE_PRL_FILE_LOCATION_RELEASE = $${CMAKE_QT_STEM}.prl
+-        } else {
+-            CMAKE_IMPLIB_FILE_LOCATION_DEBUG = $${CMAKE_QT_STEM}d.lib
+-            CMAKE_IMPLIB_FILE_LOCATION_RELEASE = $${CMAKE_QT_STEM}.lib
+-        }
++        CMAKE_IMPLIB_FILE_LOCATION_DEBUG = $${CMAKE_QT_STEM}d.lib
++        CMAKE_IMPLIB_FILE_LOCATION_RELEASE = $${CMAKE_QT_STEM}.lib
+     }
+ } else {
+     !isEmpty(CMAKE_STATIC_TYPE) {
+-- 
+2.7.4
+

--- a/src/qtbase-test.cmake
+++ b/src/qtbase-test.cmake
@@ -1,0 +1,19 @@
+# This file is part of MXE. See LICENSE.md for licensing information.
+
+# partial module - included by src/cmake/CMakeLists.txt
+
+set(TGT test-${PKG}-cmake)
+
+enable_language(CXX)
+
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
+
+find_package(Qt5 ${PKG_VERSION} EXACT REQUIRED COMPONENTS Widgets)
+
+add_executable(${TGT} ${CMAKE_CURRENT_LIST_DIR}/qt-test.cpp)
+
+target_link_libraries(${TGT} Qt5::Widgets)
+
+install(TARGETS ${TGT} DESTINATION bin)

--- a/src/qtbase.mk
+++ b/src/qtbase.mk
@@ -96,8 +96,9 @@ define $(PKG)_BUILD
         -I'$(1)/test-$(PKG)-pkgconfig' \
         `'$(TARGET)-pkg-config' Qt5Widgets$(BUILD_TYPE_SUFFIX) --cflags --libs`
 
-    # setup cmake toolchain
+    # setup cmake toolchain and test
     echo 'set(CMAKE_SYSTEM_PREFIX_PATH "$(PREFIX)/$(TARGET)/qt5" ${CMAKE_SYSTEM_PREFIX_PATH})' > '$(CMAKE_TOOLCHAIN_DIR)/$(PKG).cmake'
+    $(CMAKE_TEST)
 
     # batch file to run test programs
     (printf 'set PATH=..\\lib;..\\qt5\\bin;..\\qt5\\lib;%%PATH%%\r\n'; \

--- a/src/qtbase.mk
+++ b/src/qtbase.mk
@@ -9,7 +9,7 @@ $(PKG)_CHECKSUM := 4ef921c0f208a1624439801da8b3f4344a3793b660ce1095f2b7f5c4246b9
 $(PKG)_SUBDIR   := $(PKG)-everywhere-src-$($(PKG)_VERSION)
 $(PKG)_FILE     := $(PKG)-everywhere-src-$($(PKG)_VERSION).tar.xz
 $(PKG)_URL      := https://download.qt.io/official_releases/qt/5.14/$($(PKG)_VERSION)/submodules/$($(PKG)_FILE)
-$(PKG)_DEPS     := cc dbus fontconfig freetds freetype harfbuzz jpeg libmysqlclient libpng openssl pcre2 postgresql sqlite zlib zstd jasper libmng tiff libwebp lcms lzma
+$(PKG)_DEPS     := cc dbus fontconfig freetds freetype harfbuzz jpeg libmysqlclient libpng openssl pcre2 postgresql sqlite zlib zstd
 $(PKG)_DEPS_$(BUILD) :=
 $(PKG)_TARGETS  := $(BUILD) $(MXE_TARGETS)
 

--- a/src/qtbase.mk
+++ b/src/qtbase.mk
@@ -9,7 +9,7 @@ $(PKG)_CHECKSUM := 4ef921c0f208a1624439801da8b3f4344a3793b660ce1095f2b7f5c4246b9
 $(PKG)_SUBDIR   := $(PKG)-everywhere-src-$($(PKG)_VERSION)
 $(PKG)_FILE     := $(PKG)-everywhere-src-$($(PKG)_VERSION).tar.xz
 $(PKG)_URL      := https://download.qt.io/official_releases/qt/5.14/$($(PKG)_VERSION)/submodules/$($(PKG)_FILE)
-$(PKG)_DEPS     := cc dbus fontconfig freetds freetype harfbuzz jpeg libmysqlclient libpng openssl pcre2 postgresql sqlite zlib zstd
+$(PKG)_DEPS     := cc dbus fontconfig freetds freetype harfbuzz jpeg libmysqlclient libpng openssl pcre2 postgresql sqlite zlib zstd jasper libmng tiff libwebp lcms lzma
 $(PKG)_DEPS_$(BUILD) :=
 $(PKG)_TARGETS  := $(BUILD) $(MXE_TARGETS)
 

--- a/src/qtbase.mk
+++ b/src/qtbase.mk
@@ -105,14 +105,6 @@ define $(PKG)_BUILD
      printf 'test-qt5.exe\r\n'; \
      printf 'test-qtbase-pkgconfig.exe\r\n';) \
      > '$(PREFIX)/$(TARGET)/bin/test-qt5.bat'
-
-    # add libs to CMake config of Qt5Core to fix static linking
-    $(if $(BUILD_STATIC), \
-        $(SED) -i 's^set(_Qt5Core_LIB_DEPENDENCIES \"\")^set(_Qt5Core_LIB_DEPENDENCIES \"ole32;uuid;ws2_32;advapi32;shell32;user32;kernel32;mpr;version;winmm;z;pcre2-16;netapi32;userenv;zstd\")^g' '$(PREFIX)/$(TARGET)/qt5/lib/cmake/Qt5Core/Qt5CoreConfig.cmake' && \
-        $(SED) -i 's^set(_Qt5Gui_LIB_DEPENDENCIES \"Qt5::Core\")^set(_Qt5Gui_LIB_DEPENDENCIES \"Qt5::Core;dwmapi;winspool;wtsapi32;jasper;mng;tiff;webpdemux;webpmux;d3d11;dxgi;dxguid;harfbuzz;cairo;gobject-2.0;fontconfig;freetype;usp10;msimg32;pixman-1;ffi;expat;bz2;png16;harfbuzz_too;freetype_too;glib-2.0;shlwapi;pcre;intl;iconv;gdi32;comdlg32;oleaut32;imm32;opengl32;mpr;userenv;version;pcre2-16;netapi32;ws2_32;advapi32;kernel32;ole32;shell32;uuid;user32;winmm;lcms2;pthread;webp;zstd;lzma;jpeg;z;m\")^g' '$(PREFIX)/$(TARGET)/qt5/lib/cmake/Qt5Gui/Qt5GuiConfig.cmake' && \
-        $(SED) -i 's^set(_Qt5Network_LIB_DEPENDENCIES \"Qt5::Core\")^set(_Qt5Network_LIB_DEPENDENCIES \"Qt5::Core;ssl;crypto;ws2_32;gdi32;crypt32;iphlpapi\")^g' '$(PREFIX)/$(TARGET)/qt5/lib/cmake/Qt5Network/Qt5NetworkConfig.cmake' && \
-        $(SED) -i 's^set(_Qt5Widgets_LIB_DEPENDENCIES \"Qt5::Gui;Qt5::Core\")^set(_Qt5Widgets_LIB_DEPENDENCIES \"Qt5::Gui;Qt5::Core;gdi32;comdlg32;oleaut32;imm32;opengl32;png16;harfbuzz;ole32;uuid;ws2_32;advapi32;shell32;user32;kernel32;mpr;version;winmm;z;pcre2-16;shell32;uxtheme;dwmapi\")^g' '$(PREFIX)/$(TARGET)/qt5/lib/cmake/Qt5Widgets/Qt5WidgetsConfig.cmake',
-    )
 endef
 
 define $(PKG)_BUILD_$(BUILD)

